src/main/java/me/rigamortis/seppuku/impl/module/misc/HotBarRefillModule.java

```diff
@@ -0,0 +1,185 @@
+        for (int i = includeHotbar ? 0 : 9; i < player.inventory.mainInventory.size(); i++) {
+            ItemStack stack = player.inventory.mainInventory.get(i);
+
+            if (stack.getItem() != Items.AIR
+                && stack.getItem() == item.getItem()
+                && stack.getCount() > maxCount) {
+
+                maxCount = stack.getCount();
+                maxIndex = i;
+            }
+        }
+
+        return maxIndex;
+    }
+
+    /**
+     * Refills a given slot in the hotbar from an item in the player's inventory.
+     * Uses the slot's current ItemStack to decide what it should be refilled with.
+     *
+     * @param mc The passed Mincraft instance
+     * @param slot The slot that should be refilled
+     */
+    public void refillHotbarSlot(Minecraft mc, int slot) {
+        ItemStack stack;
+        if (slot == 45) { // Special case for offhand
+            stack = mc.player.getHeldItemOffhand();
+        } else {
+            stack = mc.player.inventory.mainInventory.get(slot);
+        }
+
+        // If the slot is air it cant be refilled
+        if (stack.getItem() == Items.AIR) {
+            return;
+        }
+
+        // The slot can't be refilled if there is nothing to refill it with
+        int biggestStack = getBiggestStack(mc.player, stack, false);
+        if (biggestStack == -1) {
+            return;
+        }
+
+        // Special case for offhand (can't use QUICK_CLICK)
+        if (slot == 45) {
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, 45, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            return;
+        }
+
+        int overflow = -1; // The slot a shift click will overflow to
+        for (int i = 0; i < 9 && overflow == -1; i++) {
+            if (mc.player.inventory.mainInventory.get(i).getItem() == Items.AIR) {
+                overflow = i;
+            }
+        }
+
+
+        if (overflow == -1) { // If the hotbar is full we can just shift click biggestStack to refill slot
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.QUICK_MOVE, mc.player);
+        } else { // If the hotbar isn't full, we might have to click the overflow stack back into the inventory
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.QUICK_MOVE, mc.player);
+
+            // If the two stacks don't overflow when combined we don't have to move overflow
+            if (mc.player.inventory.mainInventory.get(overflow).getItem() != Items.AIR) {
+                mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, overflow, ClickType.SWAP, mc.player);
```

Comment:
`windowClick(windowId, slotId, mouseButton, type, player)`

These are the game's parameter names for the windowClick(...) function, you're passing the overflow slot value for the mouseButton argument?  Are you sure this code works, have you tested it in different conditions with different slots?

Reply:
Yeah I've tested it. `ClickType.SWAP` is just when the player has their cursor on a slot in the inventory, and they press one of their hotbar keys. `windowClick()` takes `mouseButton` as the parameter for what hotbar key the player is pressing.

So in this case the situation would be that the player is hovering over the slot that they just shift-clicked the item from, and they would press the hotbar key corresponding to overflow where the items overflow to in order to return them to their original slot.